From 2a8f98d5929814225aacb411873f5bae0607da6d Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Mon, 6 Dec 2021 12:45:49 +0530
Subject: [PATCH 15151/15333] arm64: dts: qcom: sdm845-xiaomi-beryllium-common:
 add support for tas2559 speaker

Configure TAS2559 speaker amplifier which uses i2c5 and gpi_dma0.
Also, configure the audio dai link.

Inspired by work of Joel Selvaraj <jo@jsfamily.in>

[David]
 - use defines for reset-gpios, and sorting
 - removed redundant dmas and dma-names properties
Signed-off-by: David Heidelberg <david@ixit.cz>
---
 .../qcom/sdm845-xiaomi-beryllium-common.dtsi  | 27 ++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
index 8498243fd89c..796ab30d1b85 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
@@ -299,7 +299,16 @@ &i2c5 {
 
 	status = "okay";
 
-	/* TAS2559 @ 4C */
+	tas2559_codec: tas2559@4c {
+		#sound-dai-cells = <1>;
+		compatible = "ti,tas2559";
+		reg = <0x4c>;
+
+		ti,tas2559-reset-gpio = <&tlmm 12 GPIO_ACTIVE_HIGH>;
+		ti,tas2560-reset-gpio = <&tlmm 76 GPIO_ACTIVE_HIGH>;
+		ti,tas2559-addr = <0x4c>;
+		ti,tas2560-addr = <0x4d>;
+	};
 };
 
 &lab {
@@ -511,6 +520,22 @@ cpu {
 		};
 	};
 
+	tas2559-dai-link {
+		link-name = "Primary Spkr Playback";
+
+		cpu {
+			sound-dai = <&q6afedai QUATERNARY_MI2S_RX>;
+		};
+
+		platform {
+			sound-dai = <&q6routing>;
+		};
+
+		codec {
+			sound-dai = <&tas2559_codec 0>;
+		};
+	};
+
 	slim-dai-link {
 		link-name = "SLIM Playback";
 		cpu {
-- 
2.55.0

